Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -175,6 +174,15 @@ std::optional<DetachedImageBitmap> ImageBitmap::detach()
     return DetachedImageBitmap { makeUniqueRefFromNonNullUniquePtr(WTFMove(serializedBitmap)), originClean(), premultiplyAlpha(), forciblyPremultiplyAlpha() };
 }
 
+#if USE(SKIA)
+void ImageBitmap::willTransferBetweenThreads()
+{
+    // FIXME: Try avoiding GPU-to-CPU data transfer by creating accelerated clone that would e.g. leverage underlying GL texture.
+    if (m_bitmap && m_bitmap->renderingMode() == RenderingMode::Accelerated)
+        m_bitmap = m_bitmap->clone(RenderingMode::Unaccelerated);
```

Comment:
Instead of modifying clone, we could use a new method. There's also `ImageBuffer::sinkIntoBufferForDifferentThread` that only works when the buffer has only one ref for some reason. But we could add our skia specific code there, so here we would just do:

```
m_bitmap = ImageBuffer::sinkIntoBufferForDifferentThread(WTFMove(m_bitmap));
```

and sinkIntoBufferForDifferentThread already takes care of m_bitmap being null or return the same object if we don't need to copy.